### PR TITLE
Implement audio support for OTIO export

### PR DESCRIPTION
**Pull request by robin-ynput**
## Changelog Description
When working with the new publisher, I realized we do not export audio tracks in OTIO.
This changes enable it in the AYON OTIO exporter for Flame, this allow to publish "audio" product from Flame.

Also ensure that we can imediatly query current selected sequence without using "Create".
```python
from ayon_flame.api import lib
seq = lib.get_current_sequence(lib.CTX.selection)  # work even is selection is not initialized
```

## Testing notes:
1. Build a timeline with or without audio track in Flame
2. Select some clips from the timeline
3. Open the python console and export current sequence as OTIO:
```python
import ayon_flame.api as ayfapi
from ayon_flame.otio import flame_export

sequence = ayfapi.get_current_sequence(ayfapi.CTX.selection)
otio_timeline = flame_export.create_otio_timeline(sequence)
flame_export.write_to_file(otio_timeline, "/path/to/output/timeline.otio")
```

### CODE REVIEW

File: client/ayon_flame/otio/flame_export.py [hunk 1]
```diff
@@ -541,6 +541,66 @@ def _get_segment_attributes(segment):
...
+        # or add gap if following track items are having
+        # frame range differences from each other
+        elif itemindex and clip_diff != 1:
+            add_otio_gap(
+                segment_data, otio_track, prev_item["record_out"])
+
+        # create otio clip and add it to track
+        otio_clip = create_otio_clip(segment_data)
+        log.debug("_ otio_clip: %r", otio_clip)
+        otio_track.append(otio_clip)
```
Comment (iLLiCiTiT) [inline client/ayon_flame/otio/flame_export.py:601]:
```suggestion
    prev_item_record_out = segments[0]["record_out"]
    for itemindex, segment_data in enumerate(segments):
        log.debug("_ itemindex: %d", itemindex)
        log.debug("_ segment_data: %r", segment_data)

        # calculate clip frame range difference from each other
        clip_diff = segment_data["record_in"] - prev_item_record_out

        # initial track gap
        # add gap if first track item is not starting
        # at first timeline frame
        if itemindex == 0 and segment_data["record_in"] > 0:
            add_otio_gap(segment_data, otio_track, 0)

        # inbetween clip gap
        # or add gap if following track items are having
        # frame range differences from each other
        elif itemindex and clip_diff != 1:
            add_otio_gap(
                segment_data, otio_track, prev_item["record_out"])

        # create otio clip and add it to track
        otio_clip = create_otio_clip(segment_data)
        log.debug("_ otio_clip: %r", otio_clip)
        otio_track.append(otio_clip)
        prev_item_record_out = segment_data["record_out"]
```

File: client/ayon_flame/otio/flame_export.py [hunk 2]
```diff
@@ -575,57 +635,44 @@ def create_otio_timeline(sequence):
...
-                elif itemindex and clip_diff != 1:
-                    add_otio_gap(
-                        segment_data, otio_track, prev_item["record_out"])
+    # create otio audio tracks with clips
+    for audio_track in sequence.audio_tracks:
 
-                # create otio clip and add it to track
-                otio_clip = create_otio_clip(segment_data)
-                otio_track.append(otio_clip)
+        if (
```
Comment (iLLiCiTiT) [inline client/ayon_flame/otio/flame_export.py:649]: I would revert this condition to first skip, log and continue to avoid the `else` at the end.

Comment (robin-ynput) [reply client/ayon_flame/otio/flame_export.py:645]: Good call, this is done !

**Comment by jakubjezek001** [comment]
We should probably do something about following instance attributes. They should not be visible at all. But this is cosmetic and no need to block the functioning PR.

**Pull request merged**